backend/src/main/java/kr/momo/domain/attendee/AttendeeGroup.java

```diff
@@ -0,0 +1,77 @@
+package kr.momo.domain.attendee;
+
+import static java.util.stream.Collectors.collectingAndThen;
+import static java.util.stream.Collectors.toList;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import kr.momo.exception.MomoException;
+import kr.momo.exception.code.AttendeeErrorCode;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+
+@Getter
+@EqualsAndHashCode
+public class AttendeeGroup {
+
+    private final List<Attendee> attendees;
+
+    public AttendeeGroup(List<Attendee> attendees) {
+        validateUniqueNames(attendees);
+        this.attendees = attendees;
+    }
+
+    private void validateUniqueNames(List<Attendee> attendees) {
+        if (isNotUnique(attendees)) {
+            throw new MomoException(AttendeeErrorCode.DUPLICATED_ATTENDEE_NAME);
+        }
+    }
+
+    private boolean isNotUnique(List<Attendee> attendees) {
+        return !attendees.stream()
+                .map(Attendee::name)
+                .allMatch(new HashSet<>()::add);
```

Comment:
1개의 요소 매칭시 빠르게 결과를 반환하려면 다음과 같이 변경해주어야 할 것 같아 보이네요!
```suggestion
        return attendees.stream()
                .map(Attendee::name)
                .noneMatch(new HashSet<>()::add);
```

Reply:
isUnique() 함수는 각 Attendee의 이름이 서로 중복되지 않는지 확인하는 로직입니다.
그래서 전체 값이 true인 경우, true를 반환하는 allMatch를 사용하고 부정연산자(!)를 통해 반환시켰습니다.
noneMatch의 경우 전체 값이 false인 경우 true를 반환합니다. 때문에 List.of("1", "2", "2")인 경우 true를 반환할 것 같지만 false를 반환합니다. allMatch와 완전히 반대되는 개념이 아니라서, 그대로 두어야 할 것 같습니다.

---
혹시나 몰라서 이해를 위해 설명 올립니다.
```java
List<Integer> list = List.of(1, 2, 4, 4, 5);

list.stream().allMatch(HashSet<>()::add); // (1)

list.stream().noneMatch(HashSet<>()::add); // (2)
```
*먼저, allMatch와 noneMatch 내부의 HashSet<>()::add  함수는 주어진 HashSet에 add()를 통해 값을 넣은 뒤, 해당 값이 set 내부에 존재하면 false를 반환하고 해당 값이 내부에 존재하지 않으면 true를 반환합니다.

(1) AllMatch(HashSet<>()::add)
원소 1을 넣습니다 -> add의 값은 true이기 때문에 넘어갑니다.
원소 2를 넣습니다 -> add의 값은 true이기 때문에 넘어갑니다.
원소 4를 넣습니다 -> add의 값은 true이기 때문에 넘어갑니다.
원소 4를 넣습니다 -> add의 값은 false이기 멈춥니다. 
=> allMatch의 경우 하나라도 false이면 결과 값이 flase가 되기 때문에 그 뒤의 값을 볼 필요가 없습니다. 


(2) noneMatch(HashSet<>()::add)
원소 1을 넣습니다 -> add의 값은 true이기 때문에 멈춥니다.
=> noneMatch의 경우 하나라도 true면 결과 값이 false가 되기 때문에 그 뒤의 값을 볼 필요가 없습니다.

Reply:
저도 순간 헷갈렸네요ㅋㅋㅋ 배키의 설명이 정확합니다👍